rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -793,6 +794,29 @@ def _decode_transaction(
         if process_swaps:
             events = self._process_swaps(transaction=transaction, decoded_events=events)
 
+        if (
+                tx_receipt.tx_type == 4 and
+                transaction.from_address == transaction.to_address and
+                transaction.authorization_list is not None
+        ):  # if the event is an eip-7702 transaction, we need to add an informational event.
+            if (delegated_address := transaction.authorization_list[-1].delegated_address) == ZERO_ADDRESS:  # noqa: E501
+                notes = 'Successfully revoked eip-7702 transaction authorization'
+            else:
+                notes = f'Successfully executed eip-7702 transaction by delegate {delegated_address}'  # noqa: E501
+
+            events.append(self.base.make_event(
+                tx_hash=transaction.tx_hash,
+                sequence_index=self.base.get_next_sequence_index(),
+                timestamp=transaction.timestamp,
+                event_type=HistoryEventType.INFORMATIONAL,
+                event_subtype=HistoryEventSubType.NONE,
+                asset=self.value_asset,
+                amount=ZERO,
+                location_label=transaction.from_address,
```

Comment:
mmmm I have a question. Reading about the EIP it says that one can sign the transaction and then anyone can upload it
![image](https://github.com/user-attachments/assets/7788651b-c8fb-4e3e-bc19-126a136dea8e)

is correct to use the from_address here? https://etherscan.io/tx/0x42402dcf6658abaf2c47593a7ebe1264fb2f331de918239d1717a7a9d2996abf#authorizationlist etherscan seems to have it but I'm not sure if it is the from or they use something else

Reply:
edit: I believe it is correct. It is the tx.origin but it is always a transaction `from delegator address to delegator address`